Sources/AppCoinsSDK/UI/Purchase/AdyenController/AdyenController.swift

```diff
@@ -122,34 +122,34 @@ internal class AdyenController : AdyenSessionDelegate, PresentationDelegate, Obs
                 if let transactionUID = transactionUID, let method = method {
                     successHandler(method, transactionUID)
                 } else {
-                    if let failedHandler = failedHandler { failedHandler() }
+                    if let failedHandler = failedHandler { failedHandler("Missing required parameters: transaction uid is nil or payment method is nil") }
                 }
             }
             
         // The payment was refused. The response also contains a refusal reason that indicates why it was refused.
         case .refused:
-            if let failedHandler = failedHandler { failedHandler() }
+            if let failedHandler = failedHandler { failedHandler("failedHandler: Payment refused") }
```

Comment:
For all the errors on AdyenController, try to obtain more info on the error from Adyen and if possible propagate it in the message. Do not include "failedHandler" or something similar as a description, describe the error in human format and at the end indicate the location of the error with format `at <file>:<method>`.